Vector[]       vectors
Intersection[] intersections
Barcode[]      barcodes
